### PR TITLE
Update pytest config in `pyproject.toml` with repo-review recommendations

### PR DESCRIPTION
**Pull request by lagru**
## Description

This trailing field is required according to [1]. From local testing it seemed like these options weren't picked up by pytest for some time. Which makes me curious if we even need them anymore if we didn't notice that this was broken at all!

[1] https://docs.pytest.org/en/7.1.x/reference/customize.html#pyproject-toml

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
...
```

### CODE REVIEW

**Comment by lagru** [comment]
Noticed this thanks to [repo review](https://learn.scientific-python.org/development/guides/repo-review/?repo=scikit-image%2Fscikit-image&branch=main). (Thanks @henryiii ! :pray:)

**Comment by lagru** [comment]
Seems like there's some other low hanging fruits to address https://learn.scientific-python.org/development/guides/repo-review/?repo=lagru%2Fscikit-image&branch=fix-pytest-config

File: pyproject.toml [hunk 1]
```diff
@@ -192,20 +192,16 @@ exclude = [
...
-    'time_*',
-    'test_*',
-    'peakmem_*',
-]
+[tool.pytest.ini_options]
+minversion = "6.0"
+addopts = "-ra  --strict-config  --strict-markers"
+log_cli_level = "INFO"
+xfail_strict = true
+#testpaths = ["skimage", "../../../../../benchmarks"]
```
Comment (lagru) [inline pyproject.toml:200]: Our benchmark suite is broken right now when calling pytest on it. We are treating our benchmarks very stepmotherly right now. I intend to enable this option, once we have fixed the suite.

Some random thoughts on this: The approach currently taken by `spin test` makes it more difficult to run pytest on anything that is not included in the the [`f"--rootdir={site_path}"`](https://github.com/scientific-python/spin/blob/2b928bae17461748c6c5eed564dcf77a81775cd8/spin/cmds/meson.py#L228). That may apply to tests that are not included in the distribution archive, or in our case our benchmark suite. So we get this awkward inclusion path: `../../../../../benchmarks`. In this context I played around with [meson-python's (>= 0.14.0) new editable install](https://meson-python.readthedocs.io/en/latest/how-to-guides/editable-installs.html#editable-installs). That approach seems to make a lot of `spin`'s magic unnecessary. So I am carefully optimistic that in the future we can make our dev setup and tooling a lot less confusing a simpler. :crossed_fingers:

File: pyproject.toml [hunk 2]
```diff
@@ -192,20 +192,16 @@ exclude = [
...
+python_files = ["benchmark_*.py", "test_*.py"]
+python_classes = ["Test*", "*Suite"]
+python_functions = ["time_*", "test_*", "peakmem_*"]
+filterwarnings = []
```
Comment (lagru) [inline pyproject.toml:204]: Does anyone mind if I move our custom implementation of warning filters to this configuration? 

https://github.com/scikit-image/scikit-image/blob/89fa48aabdd96d7b6ae47dd111393564b7488384/skimage/_shared/testing.py#L209-L299

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**